### ISSUE TITLE
Private class field helpers throw a confusing error when in unbound method with undefined `this`

### ISSUE THREAD

**Issue by jakebailey** [microsoft/TypeScript#57609]
### 🔎 Search Terms

this undefined bind private identifier class __classPrivateFieldSet __classPrivateFieldGet bound unbound

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8BQ0PQMQDcwgFcBTaAXmgEYAmAZgBYBudTAcyIBcA1fYgCgEoU0AE4cCwgHbR2ACwCWEAHS4eRBtAC+qLamDwJEdtABmiMtAlEA7nEQCmu-RHggiikPBa8ABqfiK2LlUBcwASZD8Ajm5CIgENb34HAHpk6AB5eAAHCAAaaABJaAATOWKJAHIjACM5CWKAQkcDI0CY4nNIttUUtIAVGWF4KwgALmgAYTAJCXgjUTBi6CzhOTx2EgBbIk3qomETIc3oaeh4aoArImAjKxl4CBJQSBhSpdmjYuvwUWg5dmazlc7k8Pm6sRC5HC4L4-AS-FQQA

### 💻 Code

```ts
class Foo {
    #value = 1234;
    getValue() { return this.#value; }
}

const foo = new Foo();

console.log(`foo.getValue() = ${foo.getValue()}`);

// Oops, I didn't bind!
const getValue = foo.getValue;

// Throws: Cannot read private member from an object whose class did not declare it
console.log(`getValue() = ${getValue()}`)

```

### 🙁 Actual behavior

Throws `Cannot read private member from an object whose class did not declare it`.

But the actual problem is that `this` is undefined and I had no idea!

### 🙂 Expected behavior

A special error for `this` when undefined; IIRC there's no possible way this would normally pass, so a special error would be a lot more clear.

**Comment by fatcerberus** [comment]
For comparison, when targeting ESNext (i.e. native classfields), the error is instead:
> `Cannot read properties of undefined (reading '#value')`

**Comment by MartinJohns** [comment]
But that's a runtime error, how would you fix this without emitting additional JavaScript?

**Comment by jakebailey** [comment]
Yeah, I wasted like 40 minutes of debugging the typescript-bot trigger code only to find that I had passed Azure Function's logging function down through the handler without binding it. An error like that would have been more helpful, though I do wish the native (and downleveled code) instead mentioned `this`...
